### ISSUE TITLE
Blank comparison - KB 100900

### ISSUE THREAD

**Issue by corycundy** [tabulartools/dax-optimizer#25]
### Describe the problem

Hello, the Blank comparison issue was returned in a DAX Optimizer run on a <> CALCULATETABLE filter parameter and it isn't clear to me if this is correct or not.

The Blank comparison remarks state "When the BLANK function is used as an operand in a binary comparison operation, the engine performs an implicit conversion to compare the computed value with both 0 and BLANK, which has a small CPU cost. Moreover, a comparison with BLANK is often incorrect as 0 or an empty string (“”) returns TRUE if compared with = to BLANK."

While I agree with the fix and the basis for it, I'm seeing some discrepancy between these remarks and dax.guide for "Equal to (=)". dax.guide states "This operator does not perform any implicit conversion between strings, numbers, and Boolean values." which contradict the remarks of the issue. This same exact statement is also within "Strictly equal to (==)" and "Not equal to (<>)" in dax.guide. All three state the same.

I'm assuming that this statement in "Equal to (=)" in dax.guide may be incorrect based on the remarks and that "Strictly equal to (==)" is correct. However, my question then would be whether "Not equal to (<>)" in dax.guide is correct or not. If it is correct, then I'm not understanding why I received the issue on CALCULATETABLE parameter of "'Date'[Fiscal Week End] <> BLANK( ),"

In summary, should this issue show up when using <>? If so, is the guidance to use NOT ( ISBLANK ( ) )? Lastly, which of the three dax.guide articles are correct since they all say that they do not perform any implicit conversions?

### Power BI

SSAS 2022

### Steps to reproduce the behavior

Create DAX similar to the following:

VAR AllFiscalWeeks =
    CALCULATETABLE(
        VALUES( 'Date'[Fiscal Week End] ),
        'Date'[Fiscal Week End] <= FiscalWeekEnd,
        **'Date'[Fiscal Week End] <> BLANK( ),**
        ALL( 'Date' )
    )

**Comment by marcosqlbi** [comment]
We probably have to improve the description.
DAX Guide references the implicit conversion between different types.
DAX Optimizer references another type of conversion required to normalize BLANK to 0 (or to empty string) so that they are considered the same value in the comparison.
What about "the engine performs a transformation to compare the computed value with both 0 and BLANK" ?

**Comment by corycundy** [comment]
If all of the dax.guide descriptions are correct, then yes, your suggestion would help clarify that this is different.

Also, since this issue shows up with <> and while the example suggests "=" to be modified to "==" or ISBLANK( ), adding another example for "<>" to use NOT ( ISBLANK ( ) ) would be helpful as well. (if that is the correct guidance of course)

Thanks!

**Comment by albertospelta** [comment]
Closing the issue as the docs has been updated with a revised description for blank comparison, and a new example for `<>` to use `NOT(ISBLANK())` has been added.
